### PR TITLE
refactor README.md

### PR DESCRIPTION
**Pull request by mixmix**
## Motivation / Approach

for this : I wanted to read this in a way which was helpful for beginners. Taking from teaching, the idea I applied was "give me just enough detail to get started, then come back around with more detail as required".

So we start out **very** simple, then put more detail in the API section.

## :fire: I also mutated the API to make things clearer

As I was updating this I stumbled across things which felt like they made telling the story _harder_.
Things like:
- API behaves differently for the root feed than for all other feeds. This means we have to write more things
- use of words which conflict with other modules in ecosystem: `feedtype` is actually more like BFE `format`

I've commented those with :fire: so we can discuss.

_Andre I could have split that out into a different PR but I was writing from the point of view of ideal end state. Lets decide what we want then mutate this PR to end up where we want to be_

### CODE REVIEW

**Comment by mixmix** [comment]
# Rendered README

https://github.com/ssbc/ssb-meta-feeds/blob/04d9b8e59636b35b852fc07d97b1fa5fac0dfdf0/README.md

File: README.md
Comment (mixmix) [inline README.md:86]: Gave the simplest most common use example I could think of.
Showed how to use the feed object thingy to publish something

File: README.md
Comment (mixmix) [inline README.md:105]: wrote and explicit signature documentation for this special usage to make it clear.
I think this should be moved down in the docs

:fire:
I disliked that the object you got back from this call of this method was different, so I made it isomorphic:
- metafeed: null
- feedpuprpose: 'root'
- feedtype: 'bendybutt-v1',
- metadata: {}

NOTE I think we should change `feedtype` to `format`

File: README.md
Comment (mixmix) [inline README.md:156]: :fire: yeah `feedformat` NOT `feedtype`

Comment (mixmix) [reply README.md:156]: oh yeah 
:fire: `id` instead of `subfeed` - the whole meta / sub thing is way too many concepts for beginners. here you can easily see `id === keys.id`. `FeedDetail.id` should be the unique identifier of the Feed we're considerring ... calling `FeedDatail.subfeed` is more like asking for this things subfeed. Very confusing

I was also tempted to change `metafeed` to `parentfeed` . More happy to leave this though

File: README.md
Comment (mixmix) [inline README.md:199]: :fire: is this true?
This current signature really suggests you can use this on other peoples feedIds too....

File: README.md
Comment (mixmix) [inline README.md:283]: Introduced this type `FeedDetail` (up for changing that name) to make it easier to talk about everywhere less verbosely.
And make it easier to update docs if that thing changes

File: README.md
Comment (staltz) [inline README.md:83]: This first `null` argument may be confusing for beginners, and it's not explained that `null` is actually a short for "root" feed. I think you could restore the old example of first creating a root metafeed and then explicitly passing that root metafeed to this first argument, when creating the subfeed. This way readme readers can understand that we're "finding something" under that root metafeed, and then "creating under that root metafeed" in case it doesn't exist.

File: README.md
Comment (staltz) [inline README.md:95]: `rootFeed` is used here but it wasn't declared anywhere above it. It's good if readme readers can just copy paste the examples and they can actually run.

File: README.md
Comment (staltz) [inline README.md:128]: Where did you find `feedtype`? I grepped ssb-meta-feeds and couldn't find it.

File: README.md
Comment (staltz) [reply README.md:156]: I gave feedback about the naming `subfeed` when the meta feeds spec was being written, I think my suggestion was `id` or `feedId`. But it ended up staying as `subfeed`, and it's part of the spec, such that [go-metafeed](https://github.com/ssbc/go-metafeed) also uses `subfeed`. Not worth changing this name, IMO.

File: README.md
Comment (staltz) [inline README.md:295]: What's the purpose of "NOTE:"? Doesn't seem like it adds to the description.

File: README.md
Comment (staltz) [inline README.md:276]: I prefer the name `visit`, because "find" is the name of the method and "visit" is the name of the argument.

File: README.md
Comment (mixmix) [reply README.md:83]: You might be right.
I was trying to think "what is the most common use case of this".
And this is it (I think), but you're right it doesn't aid in learning in the same way.

Will add back in

File: README.md
Comment (mixmix) [reply README.md:95]: In this case this example is bad too!
Should not publish about messages to rootFeed..
 Right?

File: README.md
Comment (mixmix) [reply README.md:128]: Hmmm, not sure.

Comment (mixmix) [reply README.md:128]: Yeah this is weird. I remember thinking "this shouldn't be type, it's more like bfe format..."
Then I wrote the wrong one?
Strange

File: README.md
Comment (mixmix) [reply README.md:156]: Thanks for context. Will revert given its locked in already

File: README.md
Comment (staltz) [reply README.md:95]: Well, it's publishing on the aboutMeFeed, so that's fine.

File: README.md
Comment (mixmix) [reply README.md:276]: How about isFeed
I think that's clearer

File: README.md
Comment (mixmix) [reply README.md:295]: Happy to remove. I like Note of there's a perhaps expected or important side effect or behavior

File: README.md
Comment (mixmix) [reply README.md:276]: god that's so inaccessible!
I'm going with `isFeed` - says what it is on the tin

File: README.md
Comment (mixmix) [reply README.md:108]: addressed in https://github.com/ssbc/ssb-meta-feeds/pull/62

**Comment by mixmix** [comment]
@staltz have addressed all comments. thanks for review

Do you want to re-review, or shall I merge?

**Comment by mixmix** [comment]
lol... changes README, tests fail D:

**Comment by staltz** [comment]
> have addressed all comments. thanks for review
> 
> Do you want to re-review, or shall I merge?

@mixmix Let's merge, but I'll make a follow-up PR to recover some of the text you deleted, like under the "Example usage" section, there was quite a good amount of explanation and hand-holding that shouldn't be deleted.

As a general rule that arj and I have been following, the PR author merges only after the PR reviewer has approved with a green checkmark. Any moment before that is considered "don't merge yet".

**Review by staltz — APPROVED** [review]

**Review by staltz — APPROVED** [review]

**Pull request merged**
